### PR TITLE
Take list of pass-through types from config instead of hardcoding

### PR DESCRIPTION
**Pull request by ajor**
As we now store ContainerInfo objects in OICodeGen::Config, we can not copy it any more. Change all places that took copies to take const references instead.

The copy in OICodeGen modified membersToStub, the contents of which form part of OICache's hash. However, as OICache also previously had its own copy, it would not have been OICodeGen's modifications.

### CODE REVIEW

File: oi/OICodeGen.h [hunk 1]
```diff
@@ -51,18 +51,25 @@ struct ParentMember {
...
     bool useDataSegment;
 
-    FeatureSet features{};
+    FeatureSet features;
 
-    std::set<fs::path> containerConfigPaths{};
-    std::set<std::string> defaultHeaders{};
-    std::set<std::string> defaultNamespaces{};
-    std::vector<std::pair<std::string, std::string>> membersToStub{};
+    std::set<fs::path> containerConfigPaths;
```
Comment (ajor) [inline oi/OICodeGen.h:68]: The `{}` makes no difference on these members, as class-objects are always default initialised if not initialised explicitly.

However, the error messages are a lot clearer without the `{}` and removing them allowed me to track down all the locations OICodeGen was being copied.

File: oi/OICodeGen.h [hunk 2]
```diff
@@ -51,18 +51,25 @@ struct ParentMember {
...
+    Config() = default;
+    Config(const Config& other) = delete;
+    Config& operator=(const Config& other) = delete;
+    Config(Config&& other) = default;
+    Config& operator=(Config&& other) = default;
+
     /*
      * Note: don't set default values for the config so the user gets an
      * uninitialized field" warning if they missed any.
      */
```
Comment (ajor) [inline oi/OICodeGen.h:63]: @ttreyer I don't understand what exactly this comment is getting at. I've tried a few things and can't seem to get an uninitialised field warning

File: dev.oid.toml [hunk 3]
```diff
@@ -47,3 +47,8 @@ containers = [
     "PWD/types/thrift_isset_type.toml",
     "PWD/types/weak_ptr_type.toml",
 ]
+pass_through = [
```
[the remark this answers is not in the document]
Comment (ajor) [reply dev.oid.toml:50]: This file isn't documented yet.

We've got READMEs for the [container types config format](https://github.com/facebookexperimental/object-introspection/blob/main/types/README.md) and for the [integration test format](https://github.com/facebookexperimental/object-introspection/blob/main/test/integration/README.md)

**Review by tyroguru — APPROVED** [review]
That's great Alastair. Just one point about documenting the new option which I'm sure you've got covered anyway.

**Comment by ajor** [comment]
Thanks, yeah we will document this file at some point, but I'll leave it for a separate PR

**Pull request merged**